build-coordinator/src/main/java/org/jboss/pnc/core/builder/datastore/DatastoreAdapter.java

```diff
@@ -106,11 +106,41 @@ public void storeResult(BuildTask buildTask, BuildResult buildResult) throws Dat
             log.debugf("Storing results of buildTask [%s] to datastore.", buildTask.getId());
             datastore.storeCompletedBuild(buildRecordBuilder, buildTask.getBuildRecordSetIds());
         } catch (Exception e) {
-            storeResult(buildTask, e);
+            storeErroneousResult(buildTask, e);
+        }
+    }
+
+
+    public void storeFailedResult(BuildTask buildTask, BuildResult buildResult) throws DatastoreException {
```

Comment:
Do you mean BuildCoordinator class itself or do you count DatastoreAdapter as a coordinator side too?
If it's the latter - yes they are required.